JDK-8261297: NMT: Final report should use scale 1

tstuefe opened the pull request:
If PrintNMTStatistics is specified, we write a NMT report to stdout before exiting the VM. This is useful for leak analysis.

However, the report uses the standard "K" scale, which is shouldn't, since this omits small leaks < 1K. Hence it should use scale=1. This would provide us with a precise exit report. Note that for this to work JDK-8261238 has to be fixed too, which is a separate issue.

This patch:
- changes the scale of the final report to 1
- does some minor cleanup work in the MemReporter class hierarchy (const-ifying scale and the output stream, and establishing 0 as "default scale" as to avoid having the default scale hard coded in four places.

<!-- Anything below this marker will be automatically updated, please do not edit manually! -->
---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8261297](https://bugs.openjdk.java.net/browse/JDK-8261297): NMT: Final report should use scale 1


### Reviewers
 * [Zhengyu Gu](https://openjdk.java.net/census#zgu) (@zhengyu123 - **Reviewer**) ⚠️ Review applies to d24794612c8a91173f4ec19edbbc5c4889b9ade9


### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/2450/head:pull/2450`
`$ git checkout pull/2450`

zhengyu123 commented on src/hotspot/share/services/memReporter.hpp:
```diff
@@ -39,14 +39,16 @@
-    _output = (out == NULL) ? tty : out;
-  }
+  // scale = 0 -> default scale
+  MemReporterBase(outputStream* out = NULL, size_t scale = 0) :
```
Why you want to default scale parameter to 0, instead of default_scale?

zhengyu123 commented on src/hotspot/share/services/memTracker.hpp:
```diff
@@ -116,6 +116,7 @@ class MemTracker : AllStatic {
   friend class VirtualMemoryTrackerTest;
 
  public:
+
```
Unrelated?

zhengyu123 reviewed (changes requested):
Sorry for the delay.

tstuefe commented:
> Sorry for the delay.

No problem, I just pinged you in case you had not seen this PR. Since there were so many.

As requested I wired "default_scale" directly and removed the "0" as "default". I also removed the default NULL parameter for outputStream since I found no uses of MemReporter classes without an outputStream specified. 

I also removed the superfluous newline.

zhengyu123 reviewed (approved):
Looks good and trivial

tstuefe commented:
> Looks good and trivial

Thanks!

tstuefe commented:
/integrate
